archunit/src/jdk9test/java/com/tngtech/archunit/core/importer/TestLambdaMethodCalls.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.core.importer;
+
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.domain.JavaMethod;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+import com.tngtech.archunit.lang.syntax.ArchRuleDefinition;
+
+public class TestLambdaMethodCalls {
+  @Test
+  public void testName() throws Exception {
+    JavaClasses importedClasses = new ClassFileImporter()
+        .withImportOption(location -> location.contains("TestLambdaMethodCalls.DummyClassWithLambdas"))
```

Comment:
The updated test does actually not import `com.tngtech.archunit.core.importer.TestLambdaMethodCalls$DummyClassWithLambdas` and
`com.tngtech.archunit.core.importer.TestLambdaMethodCalls$DummyClassWithLambdas$Person`.

If it did, it would still fail because `useMethodsByLambda()` is not called – unlike in your earlier implementation where the methods mutually called each other.

Reply:
But you can simply use
```java
JavaClasses importedClasses = new ClassFileImporter().importClasses(DummyClassWithLambdas.class);
```
when you can boil the essence of this test down to a single example class, e.g.
```java
    static class DummyClassWithLambdas {
        void methodCallingLambdas() {
            IntStream.range(0, 1).forEach(this::methodOnlyCalledViaMethodReference);
            IntStream.range(0, 1).forEach(i -> this.methodOnlyCalledViaLambda(i));
        }

        void methodOnlyCalledViaMethodReference(int i) {
        }

        void methodOnlyCalledViaLambda(int i) {
            methodCallingLambdas();
        }
    }
```